x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -656,7 +674,7 @@ private static IndicesPrivilegesWithOptionalRemoteClusters parseIndexWithOptiona
         String currentFieldName = null;
         String[] names = null;
         BytesReference query = null;
-        String[] privileges = null;
+        String[] privileges = predefinedPrivileges;
```

Comment:
This is my take of minimal changes to reuse existing parsing code. Please let me know if this works for you. Thanks!

Reply:
I've stared at this for a while, and I think this is the best way. That parsing code is pretty gnarly, but decidedly not something to clean up in this PR.